Mathlib/CategoryTheory/Bicategory/Adjunction.lean

```diff
@@ -292,6 +292,63 @@ end Equivalence
 
 end
 
+noncomputable
+section
+
+/-- A class giving a chosen right adjoint of a 1-morphism `left`. -/
+structure RightAdjoint (left : a ⟶ b) where
+  /-- The right adjoint to `left`. -/
+  right : b ⟶ a
+  /-- The adjunction between `left` and `right`. -/
+  adj : left ⊣ right
+
+/-- The existence of a right adjoint of `f`. -/
+class IsLeftAdjoint (left : a ⟶ b) : Prop where mk' ::
+  nonempty : Nonempty (RightAdjoint left)
+
+theorem IsLeftAdjoint.mk (adj : f ⊣ g) : IsLeftAdjoint f :=
+  ⟨⟨g, adj⟩⟩
+
+/-- Use the axiom of choice to extract a right adjoint from an `IsLeftAdjoint` instance. -/
+def getRightAdjoint (f : a ⟶ b) [IsLeftAdjoint f] : RightAdjoint f :=
+  Classical.choice IsLeftAdjoint.nonempty
+
+/-- The right adjoint of a 1-morphism. -/
+def rightAdjoint (f : a ⟶ b) [IsLeftAdjoint f] : b ⟶ a :=
+  (getRightAdjoint f).right
+
+/-- Evidence that `f⁺⁺` is a right adjoint of `f`. -/
+def rightAdjoint.adjunction (f : a ⟶ b) [IsLeftAdjoint f] : f ⊣ rightAdjoint f :=
```

Comment:
In order to have similar names in category theory and bicategory theory, I would suggest:
```suggestion
def Adjunction.ofIsLeftAdjoint (f : a ⟶ b) [IsLeftAdjoint f] : f ⊣ rightAdjoint f :=
```
Same for the other side.